calicoctl/commands/diags.go

```diff
@@ -0,0 +1,188 @@
+// Copyright (c) 2016 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package commands
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/docopt/docopt-go"
+)
+
+type regPair struct {
+	reg     string
+	replace string
+}
+
+// Diags function collects diagnostic information and logs
+func Diags(args []string) error {
+	var err error
+	doc := `Usage: 
+            calicoctl diags [--log-dir=<LOG_DIR>]
+
+    Description:
+         Save diagnostic information
+
+    Options:
+         --log-dir=<LOG_DIR>  The directory for logs [default: /var/log/calico]`
+
+	arguments, err := docopt.Parse(doc, args, true, "calicoctl", false, false)
+	if err != nil {
+		return err
+	}
+
+	runDiags(arguments["--log-dir"].(string))
+
+	return nil
+}
+
+func runDiags(logDir string) {
+	cmds := []struct {
```

Comment:
I'm not super fussed either way - but if you end up adding more information to these structures it gets a little cumbersome as an inline).  If you declared it outside the scope of the function then I'd suggest just making this a structure rather than inlining that too.